# Quick validation of expert rating statistics etc.

In [2]:
import os
import pandas as pd
import numpy as np
from tqdm import tqdm
if os.getcwd().endswith("notebooks"):
    os.chdir("../")

In [32]:
raw_scores_df = pd.read_csv("data/scores/PerformanceAssessmentData_finish_corrected_filtered.csv", sep=";")

len(raw_scores_df)

327

In [33]:
raw_scores_df['Participant Number'].unique()

array([10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26,
       27, 28, 29,  1,  2,  3,  4,  5,  6,  7,  9])

In [34]:
raw_scores_df.head()

,Zeitstempel,Participant number (XXX),Expert number,1.2 Time and Motion,1.3 Instrument Handling,1.4 Knowledge of Instruments,1.5 Flow of Procedure,1.6 Knowledge of Specific Procedure,1.7 Overall Performance,1.8 Quality of final product,...,3.8 Lag screw perpendicular to fracture/ does not displace fracture,"3.9 If not lag screw, achieves compression with plate",3.10 Safely/ effectively holds plate on bone with forceps/clamp/wire,3.11 Screws drilled using appropriate guide/ soft tissue protector,3.12 Plate appropriately secured to bone in reasonable position,3.13 all screws appropriate length,CFS Overall,Did you recognize the participant?,Participant Number,Case Type
0,2024/05/08 1:01:40 PM OESZ,101,2,9,9,9,9,9,9,9,...,1,,1,1,1,1,12,No,10,radius
1,2024/05/08 1:06:06 PM OESZ,111,2,7,7,7,7,7,7,7,...,1,,1,1,1,1,12,No,11,radius
2,2024/05/08 1:08:16 PM OESZ,121,2,7,7,7,7,7,7,7,...,1,,1,1,1,1,12,No,12,radius
3,2024/05/08 1:14:56 PM OESZ,131,2,5,5,5,5,6,6,6,...,1,,1,1,1,1,11,No,13,radius
4,2024/05/08 1:17:27 PM OESZ,141,2,10,10,10,10,10,10,10,...,1,,1,1,1,1,12,Yes,14,radius


### What is the mean error and standard deviation between expert scores (with respect to mean expert score)

**ref. paper: 4.6 ± 3.8**

**I find: 4.8 ± 3.5**     

**Why is this different !??**

there are 327 not 332 samples in my data.

also here participant 19 aparently performed all 3 tasks but only gets one expert rating for Ulna case

In [26]:
# add mean expert score per participant
mean_score_df = raw_scores_df.groupby(["Participant Number", "Case Type"])['QRS Overal'].mean().reset_index().rename(columns={"QRS Overal": "Mean Expert Score"})

raw_scores_df = raw_scores_df.merge(mean_score_df, on=["Participant Number", "Case Type"], how="left")
raw_scores_df.head()

err_series = raw_scores_df['QRS Overal'] - raw_scores_df['Mean Expert Score']
err_series.abs().describe()

count    327.000000
mean       4.833333
std        3.489540
min        0.000000
25%        2.125000
50%        4.250000
75%        7.000000
max       24.250000
dtype: float64

In [35]:
mean_score_df['Mean Expert Score'].describe()

count    84.000000
mean     50.165675
std       9.537554
min      33.750000
25%      43.187500
50%      49.375000
75%      58.875000
max      69.000000
Name: Mean Expert Score, dtype: float64

In [30]:
raw_scores_df['QRS Overal'].describe()

count    327.000000
mean      49.813456
std       11.076646
min       28.000000
25%       41.000000
50%       50.000000
75%       58.000000
max       70.000000
Name: QRS Overal, dtype: float64

In [36]:
raw_scores_df[raw_scores_df['Participant Number'] == 19]

,Zeitstempel,Participant number (XXX),Expert number,1.2 Time and Motion,1.3 Instrument Handling,1.4 Knowledge of Instruments,1.5 Flow of Procedure,1.6 Knowledge of Specific Procedure,1.7 Overall Performance,1.8 Quality of final product,...,3.8 Lag screw perpendicular to fracture/ does not displace fracture,"3.9 If not lag screw, achieves compression with plate",3.10 Safely/ effectively holds plate on bone with forceps/clamp/wire,3.11 Screws drilled using appropriate guide/ soft tissue protector,3.12 Plate appropriately secured to bone in reasonable position,3.13 all screws appropriate length,CFS Overall,Did you recognize the participant?,Participant Number,Case Type
9,2024/05/08 12:22:15 PM OESZ,191,2,5,5,5,5,5,5,5,...,0,,0,0,0,0,0,No,19,radius
69,2024/06/10 5:26:10 PM OESZ,192,2,5,6,5,6,6,5,5,...,1,,1,1,1,1,9,No,19,malleolus
125,2024/07/30 11:25:00 PM OESZ,193,1,10,10,10,9,10,10,10,...,,1,1,1,1,1,10,No,19,Ulna
173,2024/08/13 10:29:44 PM OESZ,192,1,4,4,4,4,4,4,5,...,0,,1,1,1,1,10,No,19,malleolus
211,2024/09/04 12:22:17 PM OESZ,192,4,5,5,5,6,5,5,5,...,1,,0,1,0,1,6,No,19,malleolus
232,2024/09/04 9:05:30 AM OESZ,191,4,6,6,7,5,6,6,6,...,1,,0,1,1,1,8,No,19,radius
287,2024/09/18 4:38:58 PM OESZ,191,3,5,5,6,6,5,5,5,...,0,,0,0,1,0,4,No,19,radius
291,2024/09/18 9:58:15 PM OESZ,191,1,4,4,4,4,4,4,4,...,1,,0,0,1,1,6,No,19,radius
316,2024/09/24 9:23:54 AM OESZ,192,3,5,5,5,4,5,4,4,...,0,,1,1,0,0,6,No,19,malleolus
